core/store/models/job_run.go

```diff
@@ -27,23 +29,23 @@ var (
 // JobRun tracks the status of a job by holding its TaskRuns and the
 // Result of each Run.
 type JobRun struct {
-	ID             *ID          `json:"id" gorm:"primary_key;not null"`
-	JobSpecID      *ID          `json:"jobId"`
-	Result         RunResult    `json:"result" gorm:"foreignkey:ResultID;association_autoupdate:true;association_autocreate:true"`
-	ResultID       clnull.Int64 `json:"-"`
-	RunRequest     RunRequest   `json:"-" gorm:"foreignkey:RunRequestID;association_autoupdate:true;association_autocreate:true"`
-	RunRequestID   clnull.Int64 `json:"-"`
-	Status         RunStatus    `json:"status" gorm:"default:'unstarted'"`
-	TaskRuns       []TaskRun    `json:"taskRuns"`
-	CreatedAt      time.Time    `json:"createdAt"`
-	FinishedAt     null.Time    `json:"finishedAt"`
-	UpdatedAt      time.Time    `json:"updatedAt"`
-	Initiator      Initiator    `json:"initiator" gorm:"foreignkey:InitiatorID;association_autoupdate:false;association_autocreate:false"`
-	InitiatorID    int64        `json:"-"`
-	CreationHeight *utils.Big   `json:"creationHeight"`
-	ObservedHeight *utils.Big   `json:"observedHeight"`
-	DeletedAt      null.Time    `json:"-"`
-	Payment        *assets.Link `json:"payment,omitempty"`
+	ID             *ID            `json:"id" gorm:"type:uuid;primary_key;not null"`
+	JobSpecID      *ID            `json:"jobId" gorm:"type:uuid"`
+	Result         RunResult      `json:"result" gorm:"foreignkey:ResultID;association_autoupdate:true;association_autocreate:true"`
+	ResultID       clnull.Int64   `json:"-"`
+	RunRequest     RunRequest     `json:"-" gorm:"foreignkey:RunRequestID;association_autoupdate:true;association_autocreate:true"`
+	RunRequestID   clnull.Int64   `json:"-"`
+	Status         RunStatus      `json:"status" gorm:"default:'unstarted'"`
+	TaskRuns       []TaskRun      `json:"taskRuns" gorm:"foreignKey:JobRunID"`
+	CreatedAt      time.Time      `json:"createdAt"`
+	FinishedAt     null.Time      `json:"finishedAt"`
+	UpdatedAt      time.Time      `json:"updatedAt"`
+	Initiator      Initiator      `json:"initiator" gorm:"foreignkey:InitiatorID;association_autoupdate:false;association_autocreate:false"`
```

Comment:
I wonder if we can disable association auto_updates globally?

Reply:
auto_update associations are disabled globally by default, auto_create associations are not and Omit appears to be the only way